manifests/install.pp

```diff
@@ -18,7 +18,7 @@
   $python = $::python::version ? {
     'system' => 'python',
     'pypy'   => 'pypy',
-    default  => "${python::version}",
+    default  => "${python::version}", # lint:ignore:only_variable_string
```

Comment:
And doing it another way just to demonstrate the "universality" of how puppet likes to put things into their little boxes:

```
root@testbox:~# cat test.pp 
$version_int=123
$version_quoted = "${version_int}"
warning(type($version_int))
warning(type($version_quoted))
```

```
root@testbox:~# puppet apply test.pp 
Warning: Scope(Class[main]): integer
Warning: Scope(Class[main]): integer
Notice: Compiled catalog for cryptkcoding.com in environment production in 0.02 seconds
Notice: Finished catalog run in 0.30 seconds
```

In order to actually cast it into a string, you would need to do something weird like this:

```
root@testbox:~# cat test2.pp 
$version=123
$version_str = sprintf('"%s"', $version)
warning(type($version_str))
```

```
root@cryptkcoding:~# puppet apply test2.pp 
Warning: Scope(Class[main]): string
```

But again, doing that with sprintf would actually change the current behavior of the module as the current behavior is to treat the $python::version variable as either an int or a str depending on it's contents.  Removing the quotes and curly braces will resolve the lint issue while preserving that behavior.